pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -160,41 +178,69 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 		return nil
 	}
 
-	checks, err := c.podNetworkConnectivityCheckFn(ctx, syncContext)
+	var existingChecks []*v1alpha1.PodNetworkConnectivityCheck
+	if c.checkLister != nil {
+		existingChecks, err = c.checkLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+	}
+
+	newChecks, err := c.podNetworkConnectivityCheckFn(ctx, syncContext)
 	if err != nil {
 		return err
 	}
 
 	pnccClient := c.operatorcontrolplaneClient.ControlplaneV1alpha1().PodNetworkConnectivityChecks(c.namespace)
-	for _, check := range checks {
-		existing, err := pnccClient.Get(ctx, check.Name, metav1.GetOptions{})
+	newCheckNames := sets.NewString()
+	for _, newCheck := range newChecks {
+		if newCheck.Labels == nil {
+			newCheck.Labels = make(map[string]string)
+		}
+		newCheck.Labels[managedByLabelKey] = managedByLabelValue
+		newCheckNames.Insert(newCheck.Name)
+		existing, err := pnccClient.Get(ctx, newCheck.Name, metav1.GetOptions{})
 		if err == nil {
-			if equality.Semantic.DeepEqual(existing.Spec, check.Spec) {
+			if value, ok := existing.Labels[managedByLabelKey]; ok &&
+				value == managedByLabelValue && equality.Semantic.DeepEqual(existing.Spec, newCheck.Spec) {
 				// already exists, no changes, skip
 				continue
 			}
 			updated := existing.DeepCopy()
-			updated.Spec = *check.Spec.DeepCopy()
+			updated.Spec = *newCheck.Spec.DeepCopy()
+			updated.Labels = newCheck.Labels
```

Comment:
what if  the `existing` object has other labels, should we add the new label `managedByLabelKey` to `Labels` instead? that seems a safer operation.

Reply:
yes, that's correct. changed it now.